src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -1050,6 +1057,24 @@ private void EraseNodes()
 
     #region Helpers
 
+    /// <summary>
+    /// Creates a Terminal logger or Console logger based on the environment.
+    /// This method is called by reflection from dotnet. Do not modify the name or parameters without adapting the SDK.
+    /// </summary>
+    public static ILogger CreateTerminalOrConsoleLogger(LoggerVerbosity verbosity)
```

Comment:
One missing piece here is that while this does check the environment variable, the CLI flags (`--tl:off`, etc) are not respected. We may need a way to pass in an argv and have this method check for the presence/absence of the TL-related (and --console logger parameter!) flags to ensure uniform behavior.